### DISCUSSION TITLE
Executable for JavaFX Modular CLI sample application built on Windows 11 with GraalVM 22.0.1+8.1 failed to run

### DISCUSSION THREAD

**Discussion by OlexYarm** [oracle/graal#9166] [Q&A]
I built executable (native image) on Windows 11 for JavaFX Modular CLI sample application with Oracle GraalVM 22.0.1+8.1 (build 22.0.1+8-jvmci-b01).
I failed to run with error below:
Exception in thread "main" java.lang.RuntimeException: java.lang.ClassNotFoundException: hellofx.HelloFX
        at javafx.graphics@22/javafx.application.Application.launch(Application.java:311)
        at hellofx/hellofx.HelloFX.main(HelloFX.java:23)
Caused by: java.lang.ClassNotFoundException: hellofx.HelloFX
        at org.graalvm.nativeimage.builder/com.oracle.svm.core.hub.ClassForNameSupport.forName(ClassForNameSupport.java:143)
        at org.graalvm.nativeimage.builder/com.oracle.svm.core.hub.ClassForNameSupport.forName(ClassForNameSupport.java:106)
        at java.base@22.0.1/java.lang.Class.forName(DynamicHub.java:1387)
        at java.base@22.0.1/java.lang.Class.forName(DynamicHub.java:1377)
        at javafx.graphics@22/javafx.application.Application.launch(Application.java:299)
        ... 1 more

Step to reproduce
1. Install required software (Oracle GraalVM 22.0.1+8.1, OpenJFX 22 for Windows x64, Git 2.45.x, Git Desktop 2.50.x).
2. Open Windows Command Prompt
3. Clone JavaFX samples from GitHub (with Git or Git Desktop)
cd /
gh repo clone openjfx/samples
4. Build JavaFX Modular CLI sample application according to README.md
cd CommandLine\Modular\CLI\hellofx
set PATH_TO_FX="C:\openjfx-22_windows-x64_bin-sdk\javafx-sdk-22\lib"
dir /s /b src\*.java > sources.txt & javac --module-path %PATH_TO_FX% -d mods/hellofx @sources.txt & del sources.txt
5. Run sample application for testing
java --module-path "%PATH_TO_FX%;mods" -m hellofx/hellofx.HelloFX
5. Run sample with GraalVM agent for creating reflection, etc. config files
java -agentlib:native-image-agent=config-output-dir=native-image --module-path "%PATH_TO_FX%;mods" -m hellofx/hellofx.HelloFX
6. Build native image with native-image tool
native-image --verbose --native-image-info -H:+UnlockExperimentalVMOptions -H:+BuildReport -g --no-fallback --module-path "%PATH_TO_FX%;mods" -m hellofx/hellofx.HelloFX --class-path "mods" -o jfxHello
7. Run native image
jfxHello.exe

**Comment by alina-yur** [comment]
converting this discussion to an issue

**Comment by mikehearn** [comment]
Hi @OlexYarm, I know this thread is old but for others: to build JavaFX apps with native-image you have to use the Gluon Substrate project:

https://github.com/gluonhq/substrate

Unfortunately, out of the box native-image doesn't work with JavaFX.

**Reply by OlexYarm** [reply]
Hi @mikehearn, thanks for the advise. I tested gluonfx-maven-plugin long ago and it did not work with Maven 3.9.x.
I tested again latest up to date plugin version 1.0.23 and it still does not work with Maven 3.9.7:
...
Downloaded from central: https://repo.maven.apache.org/maven2/com/gluonhq/substrate/0.0.63/substrate-0.0.63.jar (1.2 MB at 6.3 MB/s)
org.apache.maven.plugin.MojoExecutionException: Maven version 3.9.7 is not currently supported by the GluonFX Maven Plugin.
Please downgrade your Maven version to 3.8.8 and then try again.
I don't want to go back in time and use outdated version of Maven.
I hope GraalVM Dev team is capable to fix a bug in GraalVM and talk to Gluon Dev team if help needed.
